hoomd/variant/box.py

```diff
@@ -0,0 +1,132 @@
+# Copyright (c) 2009-2023 The Regents of the University of Michigan.
+# Part of HOOMD-blue, released under the BSD 3-Clause License.
+
+"""Implement variants that return box parameters as a function of time."""
+
+from hoomd import _hoomd, Box
+from hoomd.data.typeconverter import box_preprocessing, variant_preprocessing
+
+
+class BoxVariant(_hoomd.VectorVariantBox):
+    """Box-like vector variant base class.
+
+    `hoomd.variant.box.BoxVariant` provides an interface to length-6 vector
+    variants that are valid `hoomd.box.box_like` objects.  The return value of
+    the ``__call__`` method returns a length-6 array of scalar values that
+    represent the quantities ``Lx``, ``Ly``, ``Lz``, ``xy``, ``xz``, and ``yz``
+    of a simulation box.
+    """
+    pass
+
+
+class Constant(_hoomd.VectorVariantBoxConstant, BoxVariant):
+    """A constant box variant.
+
+    Args:
+        box (hoomd.box.box_like): The box.
+
+    `Constant` returns ``[box.Lx, box.Ly, box.Lz, box.xz, box.xz, box.yz]`` at
+    all time steps.
+
+    """
+
+    def __init__(self, box):
+        box = box_preprocessing(box)
+        BoxVariant.__init__(self)
+        _hoomd.VectorVariantBoxConstant.__init__(self, box._cpp_obj)
+
+    @property
+    def box(self):
+        """hoomd.Box: The box."""
+        return Box._from_cpp(self._box)
+
+    @box.setter
+    def box(self, box):
+        box = box_preprocessing(box)
+        self._box = box._cpp_obj
+
+
+class Interpolate(_hoomd.VectorVariantBoxInterpolate, BoxVariant):
+    """Interpolate between two boxes linearly in time.
+
+    Args:
+        initial_box (hoomd.box.box_like): The initial box.
+        final_box (hoomd.box.box_like): The final box.
+        variant (hoomd.variant.variant_like): A variant used to interpolate
+            between the two boxes.
+
+    `Interpolate` returns the array corresponding to *initial_box* for
+    :math:`t \\leq t_{\\mathrm{start}}` and *final_box* for
+    :math:`t \\geq t_{\\mathrm{start}} + t_{\\mathrm{ramp}}`.
```

Comment:
This is incorrect, there is no t_start nor t_ramp. Replace this with a copy of the docs from `BoxResize` that explicitly document the interpolation.